cmd/scaffold.go

```diff
@@ -187,6 +190,15 @@ func (fnb *FlowNodeBuilder) BaseFlags() {
 	fnb.flags.UintVar(&fnb.BaseConfig.SyncCoreConfig.MaxRequests, "sync-max-requests", defaultConfig.SyncCoreConfig.MaxRequests, "the maximum number of requests we send during each scanning period")
 
 	fnb.flags.Uint64Var(&fnb.BaseConfig.ComplianceConfig.SkipNewProposalsThreshold, "compliance-skip-proposals-threshold", defaultConfig.ComplianceConfig.SkipNewProposalsThreshold, "threshold at which new proposals are discarded rather than cached, if their height is this much above local finalized height")
+
+	// unicast stream handler rate limits
+	fnb.flags.IntVar(&fnb.BaseConfig.UnicastMessageRateLimit, "unicast-message-rate-limit", 0, "amount of unicast messages that sent by a peer per second")
+	fnb.flags.IntVar(&fnb.BaseConfig.UnicastMessageBurstLimit, "unicast-message-burst-limit", 1, "amount of unicast messages that can be sent by a peer at one time")
```

Comment:
See the rate library [implementation](https://cs.opensource.google/go/x/time/+/master:rate/rate.go;l=371) for how the burst value is used
> // ... Burst is the maximum number of tokens
// that can be consumed in a single call to Allow, Reserve, or Wait, so higher
// Burst values allow more events to happen at once.

Reply:
When rate limiting the messages we need a 1:1 ratio between burst and limit. Burst is the rate at which the tokens are replenished in the token bucket https://en.wikipedia.org/wiki/Token_bucket . For some reason with golang's implementation if we set burst to 1 it will only allow 1 message.

Reply:
I removed the burst config and now set the burst to the limit

Reply:
https://github.com/onflow/flow-go/pull/2840/commits/6f35feb404725027f593efa63a21cb19cd3a6418